infrastructure/resource_group_level/modules/trade-finance-manager-api.bicep

```diff
@@ -0,0 +1,251 @@
+  // APPLICATIONINSIGHTS_CONNECTION_STRING: applicationInsights.properties.ConnectionString
+  
+  DOCKER_ENABLE_CI: 'true'
+  DOCKER_REGISTRY_SERVER_URL: '${containerRegistryName}.azurecr.io'
+  DOCKER_REGISTRY_SERVER_USERNAME: dockerRegistryServerUsername
+  LOG4J_FORMAT_MSG_NO_LOOKUPS: 'true'
+  TZ: 'Europe/London'
+  WEBSITE_HTTPLOGGING_RETENTION_DAYS: '3'
+  WEBSITES_ENABLE_APP_SERVICE_STORAGE: 'false'
+  
+  // TODO:FN-737 The following 4 secrets are from the APIM changes
+  // Note that DTFS_CENTRAL_API_URL & EXTERNAL_API_URL are also in Connection Strings! (via CLI)
+  DTFS_CENTRAL_API_URL: dtfsCentralApiUrl
+  EXTERNAL_API_URL: externalApiUrl
+  PORTAL_API_URL: portalApiUrl
+  // TODO:FN-737 - This is a self-reference. I don't think it is needed. Double check and remove.
+  TFM_API_URL: 'TODO:FN-737' // https://tfs-dev-trade-finance-manager-api.azurewebsites.net/
+}
+
+var nodeEnv = environment == 'dev' ? { NODE_ENV: 'development' } : {}
+
+var appSettings = union(settings, secureSettings, additionalSettings, additionalSecureSettings, nodeEnv)
+
+var tfmApiName = 'tfs-${environment}-trade-finance-manager-api'
+var privateEndpointName = 'tfs-${environment}-trade-finance-manager-api'
+var applicationInsightsName = 'tfs-${environment}-trade-finance-manager-api'
+
+
+var connectionStringsList = [for item in items(union(secureConnectionStrings, additionalSecureConnectionStrings)): {
+  name: item.key
+  value: item.value
+} ]
+
+var connectionStringsProperties = toObject(connectionStringsList, item => item.name, item => {
+  type: 'Custom'
+  value: item.value
+} )
+
+
+// Then there are the calculated values. 
+var mongoDbConnectionString = replace(cosmosDbAccount.listConnectionStrings().connectionStrings[0].connectionString, '&replicaSet=globaldb', '')
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2022-09-01' existing = {
+  name: storageAccountName
+}
+var storageAccountKey = storageAccount.listKeys().keys[0].value
+
+var connectionStringsCalculated = {  
+  MONGO_INITDB_DATABASE: {
+    type: 'Custom'
+    value: cosmosDbDatabaseName
+  }
+  MONGODB_URI: {
+    type: 'Custom'
+    value: mongoDbConnectionString
+  }
+  EXTERNAL_API_URL: {
+    type: 'Custom'
+    value: externalApiUrl
+  }
+  // TODO:FN-737 Note that DTFS_CENTRAL_API becomes DTFS_CENTRAL_API_URL
+  DTFS_CENTRAL_API: {
+    type: 'Custom'
+    value: dtfsCentralApiUrl
+  }
+  AZURE_NUMBER_GENERATOR_FUNCTION_URL: {
+    type: 'Custom'
+    value: 'TODO:FN-420'
+  }
+} 
+
+
+var connectionStringsCombined = union(connectionStringsProperties, connectionStringsCalculated)
+
+
+resource cosmosDbAccount 'Microsoft.DocumentDB/databaseAccounts@2023-04-15' existing = {
+  name: cosmosDbAccountName
+}
+
+resource tfmApi 'Microsoft.Web/sites@2022-09-01' = {
+  name: tfmApiName
+  location: location
+  tags: {
+    Environment: 'Preproduction'
+  }
+  kind: 'app,linux,container'
+  properties: {
+    httpsOnly: false
```

Comment:
we can leave it as `false` for now. but we need to look at setting it to `true` afterwards. Not sure if we need to change anything else

Reply:
Apart from making sure clients are updated to use https, I don't think there's anything else. I'm mentioned both steps in FN-741.